inference-engine/tests/functional/shared_test_classes/src/single_layer/detection_output.cpp

```diff
@@ -64,10 +64,36 @@ std::string DetectionOutputLayerTest::getTestCaseName(const testing::TestParamIn
     return result.str();
 }
 
-void DetectionOutputLayerTest::GenerateInputs() {
-    size_t it = 0;
-    for (const auto &input : cnnNetwork.getInputsInfo()) {
-        const auto &info = input.second;
+ParamsWhichSizeDepends DetectionOutputLayerTest::fromStatic(
+    const bool varianceEncodedInTarget,
+    const bool shareLocation,
+    const bool normalized,
+    const size_t inputHeight,
+    const size_t inputWidth,
+    const ov::Shape& locationInput,
+    const ov::Shape& confidenceInput,
+    const ov::Shape& priorsInput,
+    const ov::Shape& armConfidence,
+    const ov::Shape& armLocation) {
+    return {
+        varianceEncodedInTarget,
+        shareLocation,
+        normalized,
+        inputHeight,
+        inputWidth,
+        {locationInput, {locationInput}},
+        {confidenceInput, {confidenceInput}},
+        {priorsInput, {priorsInput}},
+        {armConfidence, {armConfidence}},
+        {armLocation, {armLocation}}
+    };
+}
+
+void DetectionOutputLayerTest::generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) {
+    inputs.clear();
+    const auto& funcInputs = function->inputs();
+    for (auto it = 0ul; it < funcInputs.size(); ++it) {
```

Comment:
```suggestion
    for (auto i = 0ul; i < funcInputs.size(); ++i) {
```

Otherwise it looks like an iterator.

Reply:
done